src/index.tsx

```diff
@@ -119,23 +119,39 @@ startMathJax();
 // load optimize
 loadOptimize(window, app.store);
 
-// Learn more about service workers: http://bit.ly/CRA-PWA
-serviceWorker.register()
-  .then((registration) => {
-    if (!window.navigator.serviceWorker.controller) {
-      return;
-    }
-
-    if (registration && (registration.waiting || registration.installing)) {
-      app.store.dispatch(updateAvailable());
-    } else if (registration) {
-      // For Chrome and Edge registration.waiting and registration.installing
-      // is still null for some time after .register()
-      registration.addEventListener('updatefound', () => {
+function securityIsTooTight() {
+  const inAnIframe = window.location !== window.parent.location;
+  let lsDisabled = false;
+
+  try {
+    window.localStorage.setItem('securityTest', '1');
+    window.localStorage.removeItem('securityTest');
+  } catch (e) {
+    lsDisabled = true;
+  }
+
+  return inAnIframe && lsDisabled;
+}
+
+if (!securityIsTooTight) {
+  // Learn more about service workers: http://bit.ly/CRA-PWA
+  serviceWorker.register()
```

Comment:
Do the service works in general not work if localStorage is disabled? I thought it was just going to be the text resizing that was affected

Reply:
Service workers won't initialize if two things are true: running in an iframe _and_ localstorage is not working. Or at least that's how I read the issue.

Reply:
i'm not sure if cookies specifically is the problem, but the service worker registration is getting a permission denied message, i think its happening when third party cookies are disabled

Reply:
reading the code -- i would try to recognize the security error in the catch before reporting to sentry rather than assume that the security issue is related to localStorage (which may be true but also might be coincidental)